soma-config.1
.TH soma-config 1 "9 July 2005" "soma" "soma-config manual"
.SH NAME
soma-config \- Get information about a soma installation
.SH SYNOPSIS
.B soma-config [options]
.SH DESCRIPTION
.B soma-config
displays information about a previous soma installation.
.SH OPTIONS
.IP "--prefix"
Show the prefix path of soma
.IP "--version"
Show the version of soma
.IP "--cflags"
Set of compiler options (CFLAGS) to use when compiling files that use
soma.
.IP "--libs"
Shows the complete set of libs and other linker options you will need in order
to link your application with soma.
.IP "--ssl"
Return 0 if soma has not ssl support, 1 if soma is compiled with libssl.
.SH "EXAMPLES"
What linker options do I need when I link with soma?

  soma-config --libs

What compiler options do I need when I compile using soma functions?

  soma-config --cflags
.SH SEE ALSO
.BR somad (1)
.SH AUTHOR
This manual page was written by Andrea Marchesini <bakunin@autistici.org>.


ChangeLog
18.02.2007		soma	2.4
	NextItem support
	Song -> Item
	ffmpeg as external library
	No XmlSyntax
	RandomItem and RandomSpot
	phpsoma and pysoma with enable flags
	manpages updated
	pause

19.07.2006		soma	2.3
	Tnx to klaus for a little bug in the libsoma
	Before group and after user
	31th of month

08.05.2006		soma	2.2
	a little bug fixes

27.10.2005		soma	2.1
	dtd for palinsesto and spot file
	ffmpeg for stat on the distribuited file
	hash table for stat and stat_dir
	cvs opened
	PreSpot and PostSpot

16.06.2005		soma	2.0
	Release 1.14 -> 2.0

09.06.2005		soma	1.14
	Spot Controller
	Spot file Like Palinsesto file (upload, change,...)
	New protocol layout
	Nextsong iteration with spot controller (or pathspot)
	palinsesto and spot files on xml syntax
	playlist pls and m3u support
	ssl support
	unix socket support
	PATH enviroment used for programsong and programstream.
	SymLinks function
	module_list funcion
	Deny and Allow list

08.03.2005		soma	1.13
	skip to next song
	nextsong remove before play
	ModuleData
	Set default palinsesto
	
15.02.2005		soma	1.12
	bugfixes about changes of palinsesto file
	wait after some kill

05.02.2005		soma	1.11
	autopath for nextsong set
	scandir vs readdir
	relative path to distribuited paths
	mutex bug fixed
	libtoolize

05.01.2005		soma	1.10
	configure.ac with check for MacOSX
	nextsong remove/list/insert options
	change the command in somaclient shell mode
	distribuited fs
	libsoma
	log.c day array by <megabug@autistici.org>

13.12.2004		soma	1.9
	little check in somad about ProgramSong and ProgramSpot
	bug fixed in nextsong options when it is out of palinsesto
	nextsong list

29.11.2004		soma	1.8
	another algorithm for scheduler!
	somaclient shell history
	somad: new layout of Start and Stop string

22.11.2004		soma	1.7
	new algorithm for scheduler
	bug fixed - somaclient (shell mode)
	
14.11.2004		soma	1.6
	add timer filter for star and stop trasmissions
	
22.09.2004		soma	1.4
	somacheck fixed bug

07.01.2004		soma	1.2
	debuggati i moduli e scritto somadiff.
	cambiati i segnali quando parte il programma di playing
	in modo da supportare pure mplayer.
	RatioSong e RatioSpot al posto di Ratio.
	Abbandonato i syslog per una gestione dei log interna.

07.01.2004		soma	1.0
	debuggati i moduli e scritto somadiff.
	cambiati i segnali quando parte il programma di playing
	in modo da supportare pure mplayer.
	RatioSong e RatioSpot al posto di Ratio.
	Abbandonato i syslog per una gestione dei log interna.

02.01.2004		soma	0.99
	aggiunto per ogni trasmissione il Jingle.
	nasce soma come link a somaclient.
	supporto moduli per somad.
	Migliorato il codice.
	
02.12.2003		soma	0.98
	Corretti un po' di bugs.
	Abilitazione file URL sia in PathSong e sia in PathSpot con
	lancio di ProgramStream all'occorrenza.
	Spostato Ratio dentro al palinsesto cosi' da renderlo ancor piu'
	configurabile

22.11.2003		soma	0.97
	somaclient -ppassword per utilizzarlo meglio a riga di comando
	somaclient --start e --stop per stoppare a riattivare somad.
	Commenti al codice.
	Inserita l'opzione -c in somaclient per avere i messaggi in formato
	codice (utile per un eventuale frontend).
	Ottimizzazioni nella gestione delle directory e files in somad.

16.11.2003		soma	0.96
	Makefile
	Sostanziali modifiche a somaclient aggiundo readline e
	getpalinsesto.
	Correzione bugs del palinsesto.

14.11.2003		soma	0.95
	Nuovo algoritmo del palinsesto circolare.
	Correzione bugs riguardo al socket.
	Status di uscita dei programmi.
	Documentazione in tex e html.
	Lo status di somad consultabile da somaclient.
	Debug a + livelli.
	/var/run/soma.pid.
	Il solo check della password con somaclient.
	Caricare palinsesto con somaclient o prendere quello di default.
	ServerName per il binding della connessione
	
11.11.2003		soma	0.94
	Socket TCP per l'amministrazione.
	Eliminato il sistema dei segnali e impolpito la gestione dei mutex
	per le strutture dati.
	Nasce il supporto per lo Streaming

09.11.2003		soma	0.93
	Syslogd per i log.
	Nasce somaclient, somacheck e soma diventa somad.
	Palinsesto con PathSong di default + per ogni trasmissione. Idem
	per le PathSpot e Random.

31.09.2003 		soma	0.92
	Gestione del file attraverso libconfuse.
	Gestione di un palinsteso.
	Divisione in + file.
	Implementazione prima di uno unix socket per la gestione del demone.

	
30.09.2003		soma	0.91
	Gestione di + PathSong e + PathSpot. Aggiunta di Background, Random,
	Program e Options nel file di configurazione.
	Random a richiesta.
	Messo on line su http://autistici.org/bakunin
	Supporto di Directory e di File come PathSong e PathSport


config/soma-config.c
/* Soma - Copyright (C) 2003-7 bakunin - Andrea Marchesini 
 *                                       <bakunin@autistici.org>
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  
 * 02110-1301, USA.
 *
 * This program is released under the GPL with the additional exemption that
 * compiling, linking, and/or using OpenSSL is allowed.
 */

#ifdef HAVE_CONFIG_H
#include <config.h>
#else
# error Use configure; make; make install
#endif

#include <stdio.h>
#include <string.h>

int
usage (void)
{
  fprintf (stderr, "Usage: soma-config [OPTIONS]\nOptions:\n");
  fprintf (stderr, "\t[--prefix]\n");
  fprintf (stderr, "\t[--version]\n");
  fprintf (stderr, "\t[--libs]\n");
  fprintf (stderr, "\t[--cflags]\n");
  fprintf (stderr, "\t[--ssl]\n");

  return 1;
}

int
main (int argc, char *argv[])
{
  register int i;

  if (argc < 2)
    return usage ();

  for (i = 1; i < argc; i++)
    {
      if (strcmp (argv[i], "--prefix") &&
	  strcmp (argv[i], "--version") &&
	  strcmp (argv[i], "--libs") &&
	  strcmp (argv[i], "--cflags") && 
	  strcmp (argv[i], "--ssl"))
	return usage ();
    }

  for (i = 1; i < argc; i++)
    {
      if (!strcmp (argv[i], "--prefix"))
	fprintf (stdout, "/usr");

      else if (!strcmp (argv[i], "--version"))
	fprintf (stdout, "2.4");

      else if (!strcmp (argv[i], "--libs"))
	fprintf (stdout, " -lxml2 -lz -lm  " " -lhistory -lreadline -lncurses -ldl -lpthread  -lconfuse -lssl -lcrypto  ");

      else if (!strcmp (argv[i], "--cflags"))
	fprintf (stdout, "-g -O2   -DSOMA_USE_OPENSSL -I/usr/include/libxml2   -Wall");

      else if (!strcmp (argv[i], "--ssl"))
#ifdef SOMA_USE_OPENSSL
	fprintf (stdout, "1")
#else
	fprintf (stdout, "0")
#endif
	  ;
    }

  fprintf (stdout, "\n");

  return 0;
}

/* EOF */
